inkscape-extensions.inkstitch: fix editor opening

felbinger opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

felbinger committed:
inkscape-extensions.inkstitch: fix editor opening

Co-authored-by: tropf <29873239+tropf@users.noreply.github.com>

felbinger commented:
depends on https://github.com/NixOS/nixpkgs/pull/366537, otherwise plugin will not be loaded at all in `inkscape-with-extensions` pkg

pluiedev commented:
Doesn't seem to fully work: if I click on Extensions > Ink/Stitch > About nothing shows up

tropf committed:
inkscape-extensions.inkstitch: add implicit dependencies in build

Inkstitch assumes its runtime python environment to be present during
build, so provide it.

tropf committed:
inkscape-extensions.inkstitch: provide version number

Overwrite the version number shown in inkstitch's about dialog. Without
this option, only 'manual' (without a version number) is shown.

tropf committed:
maintainer: add tropf

tropf committed:
inkscape-extensions.inkstitch: add tropf as maintainer

tropf commented:
that's on me, we didn't test properly.
The dependencies have to be present at build time, which we forgot.

The current version (with #366537 cherry-picked) works for me. Does it for you?
Tested: About, Simulator, Params view, PDF export

As I have already maintained a nixpkgs inkstitch out-of-tree for over 2 years, I have added myself to the maintainer list.
(I have no experience with this beyond reading the guide. If this is the wrong place/PR to do that, please tell me.)

SuperSandro2000 commented on pkgs/applications/graphics/inkscape/extensions/inkstitch/default.nix:
```diff
@@ -47,6 +30,26 @@ python3.pkgs.buildPythonApplication rec {
+  src = fetchFromGitHub {
+    owner = "inkstitch";
+    repo = "inkstitch";
+    rev = "refs/tags/v${version}";
```
```suggestion
    tags = "v${version}";
```

pluiedev replied:
Should be `tag`, singular, no?

tropf replied:
done in cd87669d6f040611e6e28bfd56c5d5772c40c560

SuperSandro2000 commented on pkgs/applications/graphics/inkscape/extensions/inkstitch/default.nix:
```diff
@@ -51,6 +51,11 @@ python3.pkgs.buildPythonApplication {
 
+  env = {
+    # to overwrite version string
+    GITHUB_REF = "${version}";
```
```suggestion
    GITHUB_REF = version;
```

tropf replied:
adjusted in e74bd37a8c15f1c96e8f9c8e5728438dafcce4c2

tropf committed:
inkscape-extensions.inkstitch: simplify fetchFromGitHub
